packages/typography/src/Link/Link.spec.tsx

```diff
@@ -211,6 +211,46 @@ describe('packages/typography', () => {
           <Link as="a">Content</Link>
         </>;
       });
+
+      // eslint-disable-next-line
+      test.skip('LinkProps types', () => {
+        // eslint-disable-next-line
+        const withHref: LinkProps = {
+          href: 'string',
+        };
+
+        // eslint-disable-next-line
+        const asProp: LinkProps = {
+          as: 'div',
+        };
+
+        // @ts-expect-error href is not allowed on explicit div
+        // eslint-disable-next-line
+        const asAndHref: LinkProps = {
+          as: 'div',
+          href: 'string',
+        };
+
+        // should expect error, but this does not
```

Comment:
this should error @TheSonOfThomp, but it does not. I'm not sure that we should block the PR on this but i did want to call that out and see if you had any thoughts/ideas

Reply:
The prop type util isn't quite as robust as the component type check, We can add a ticket to fix this

Reply:
https://jira.mongodb.org/browse/LG-3227